### PR TITLE
Force cmd.exe to be used by system() instead of pwsh or powershell on Windows.

### PR DESCRIPTION
**Pull request by Acelya-9028**
cf #751.

### CODE REVIEW

File: autoload/gitgutter/utility.vim [hunk 1]
```diff
@@ -176,15 +176,20 @@ endfunction
...
   if has('unix') && &shell !=# 'sh'
-    let [s:shell, s:shellcmdflag, s:shellredir] = [&shell, &shellcmdflag, &shellredir]
+    let [s:shell, s:shellcmdflag, s:shellredir, s:shellpipe, s:shellquote, s:shellxquote] = [&shell, &shellcmdflag, &shellredir, &shellpipe, &shellquote, &shellxquote]
     let &shell = 'sh'
     set shellcmdflag=-c shellredir=>%s\ 2>&1
   endif
+  if has('win32') && (&shell ==# 'pwsh' || &shell ==# 'powershell')
+    let [s:shell, s:shellcmdflag, s:shellredir, s:shellpipe, s:shellquote, s:shellxquote] = [&shell, &shellcmdflag, &shellredir, &shellpipe, &shellquote, &shellxquote]
+    let &shell = 'cmd.exe'
+    set shellcmdflag=/s\ /c shellredir=>%s\ 2>&1 shellpipe=>%s\ 2>&1 shellquote= shellxquote="
```
Comment (airblade) [inline autoload/gitgutter/utility.vim:186]: The `shellxquote` value appears to be a double-quotation mark.  Is that right?  I would have expected `'"'` or `"\""` in that case.

**Comment by airblade** [comment]
Do you have any other `shell*` options set in your vimrc, e.g. `shellcmdflag` or `noshelltemp`, which could affect this?

**Comment by Acelya-9028** [comment]
It's my config:

```
set shell=pwsh shellpipe=\| shellxquote=
set shellcmdflag=-NoLogo\ -NoProfile\ -ExecutionPolicy\ RemoteSigned\ -Command
set shellredir=\|\ Out-File\ -Encoding\ UTF8
``` 

but the official is [runtime/doc/options.txt#L5209](https://github.com/neovim/neovim/blob/fbbc8cb8b110ebf05f33dd4bce83a59b7a79d822/runtime/doc/options.txt#L5209).

File: autoload/gitgutter/utility.vim (hunk 1 above)
Comment (Acelya-9028) [reply autoload/gitgutter/utility.vim:186]: Here is my default:
shellxquote

Comment (airblade) [reply autoload/gitgutter/utility.vim:186]: Ah, I guess option values set with `:set` don't need quoting.

**Comment by airblade** [comment]
We don't need to worry about `noshelltemp` because `system()` ignores it.

It sounds like we're setting and restoring the appropriate options.

**Comment by Acelya-9028** [comment]
Thanks!

**Pull request merged**